libraries/Microsoft.Bot.Builder.AI.QnA/DialogHelper.cs

```diff
@@ -0,0 +1,188 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.AI.QnA
+{
+    /// <summary>
+    /// QnAMaker Active Learning Dialog helper class
+    /// </summary>
+    public class DialogHelper
+    {
+        /// <summary>
+        /// QnA Maker Active Learning dialog name
+        /// </summary>
+        public const string ActiveLearningDialogName = "active-learning-dialog";
+
+        /// <summary>
+        /// Gets QnA Maker Active Learning Dialog.
+        /// </summary>
+        public WaterfallDialog QnAMakerActiveLearningDialog { get; }
+        
+        // Define value names for values tracked inside the dialogs.
+        private const string CurrentQuery = "value-current-query";
+        private const string QnAData = "value-qnaData";
+
+        // Dialog Options parameters
+        private const float DefaultThreshold = 0.03F;
+        private const int DefaultTopN = 3;
+
+        // Card parameters
+        private const string CardTitle = "Did you mean:";
+        private const string CardNoMatchText = "None of the above.";
+        private const string CardNoMatchResponse = "Thanks for the feedback.";
+
+        private readonly QnAMaker _services;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DialogHelper"/> class.
+        /// Dialog helper to generate dialogs.
+        /// </summary>
+        /// <param name="services">Bot Services.</param>
+        public DialogHelper(QnAMaker services)
+        {
+            QnAMakerActiveLearningDialog = new WaterfallDialog(ActiveLearningDialogName)
```

Comment:
I'm still not sold on the need for the need for WaterfallDialog here. Can we try to use ADaptiveDialog? WE don't use adaptivedialog because you want to avoid a dependency with adaptive? I don't understand the reason behind that, please help me understand the reasoning.

Reply:
Our main point of opting for Waterfall dialog was that, it was solving our propose and the Adaptive dialog component are still under development and we have very tight timeline as I am the only dev working for all the QnAMaker connection with botframework. So, As per our last conversation and yesterday's conversation with Vishwac and Yochay, for this version we will start with Waterfall and if time persists, we will pick up Adaptive version.